splitio/client/factory.py

```diff
@@ -140,12 +153,31 @@ def _start_status_updater(self):
             ready_updater.start()
         else:
             self._status = Status.READY
+            init_updater = threading.Thread(target=self._update_redis_init,
```

Comment:
there is no need to do this operation in a thread you can execute `_update_redis_init` directly. In addition, there is no need to create a function for this

Reply:
If I execute _update_redis_init it will not get the factory active count, since the factory object init method is not finished yet. Running it in a thread will ensure it runs after the factory init finish, and will capture the active and redundant factories

Reply:
Moved record init for redis to after creating the factory instance, thread is removed.